kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/filter/EventLoopFilterThreadExecutor.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.internal.filter;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import io.netty.channel.EventLoop;
+
+import io.kroxylicious.proxy.filter.FilterThreadExecutor;
+
+public class EventLoopFilterThreadExecutor implements FilterThreadExecutor {
```

Comment:
This feels like a perfectly natural implementation! However [this change](https://github.com/kroxylicious/kroxylicious/pull/965/files#diff-4e4ada152a1206c17f9cf1382ce9c7dc9b94b39479e651a6f20acd75c77794dcL49) did make we wonder is it the right one?

The Netty event loop is a `ScheduledExecutorService` itself so is it better if this impl accepts a `ScheduledExecutorService` and delegates to it?

Reply:
To answer my own question, it is needed while we offer `isInFilterThread` as we can't call `inEventLoop` otherwise. However I'm not convinced that is needed. When is it useful? The vast majority of the time people would use it to re-implement [completedOnFilterThread]( https://github.com/kroxylicious/kroxylicious/pull/965/files#diff-88e0e0cd282e904b79116049f18eb7739750f5bc88c023fbfc42524fe6abfcd6R39)

Reply:
One reason is to make it easier to integration test our threading model guarantees, the filters can check that they 
 are being called from the Filter thread.
 
 I think the method could also be useful for writing guard clauses (sanity checks) around sensitive operations. Like having an `if (!executor.isInFilterThread()) { throw new RuntimeException("xyz")}` as an additional check might appear redundant after switching, but it could protect you from a refactor breaking things.